util/http_util.go

```diff
@@ -70,3 +75,41 @@ func CreateHTTPClient() *http.Client {
 		},
 	}
 }
+
+func CreateHTTPClientWithProxy(proxyUrl string) *http.Client {
+
+	client := CreateHTTPClient()
+	if proxyUrl == "" {
+		return client
+	}
+
+	// Proxy URL format check
+	_, err := url.ParseRequestURI(proxyUrl)
+	if err != nil {
+		log.Println("Proxy parse error, disable the proxy")
+		return client
+	}
+
+	// Set proxy url
+	client.Transport.(*http.Transport).Proxy = func(_ *http.Request) (*url.URL, error) {
+		return url.Parse(proxyUrl)
+	}
+
+	// Check if proxy is alive
+	req, err := http.NewRequest(http.MethodHead, proxyCheckUrl, nil)
+	if err != nil {
+		log.Println("Proxy test request create error, disable the proxy")
+		client.Transport.(*http.Transport).Proxy = nil
+		return client
+	}
+	resp, err := client.Do(req)
+	if err != nil || resp == nil || resp.StatusCode >= 300 {
+		// Not alive, proxy will not be set
+		log.Println("Proxy test failed (Cannot access", proxyCheckUrl, "using proxy", proxyUrl, "), disable the proxy")
+		client.Transport.(*http.Transport).Proxy = nil
+		return client
+	}
+
+	// Return HTTP client with proxy
+	return client
```

Comment:
服务器返回的状态码应始终为 204，同时，若请求成功，应始终关闭 `response.Body`：

```suggestion
	if err == nil {
		resp.Body.Close()
		if resp.StatusCode != http.StatusNoContent {
			// Return HTTP client with proxy
			return client
		}
	}
	// Not alive, proxy will not be set
	log.Println("Proxy test failed (Cannot access", proxyCheckUrl, "using proxy", proxyUrl, "), disable the proxy")
	client.Transport.(*http.Transport).Proxy = nil

	// Return HTTP client without proxy
	return client
```